app/services/activity_creator.rb

```diff
@@ -8,6 +8,32 @@ def process
       @activity.points = @activity.activity_type.score
       @activity.activity_category = @activity.activity_type.activity_category
     end
-    @activity.save
+
+    if @activity.save
+      process_programmes if started_active_programmes.any?
```

Comment:
Not sure this check is necessary given we'd be iterating through an empty list later on.